src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -178,7 +182,15 @@ internal IEnumerable<ICompletedFakeObjectCall> GetRecordedCalls()
         /// <param name="call">The call to remember.</param>
         internal void RecordCall(ICompletedFakeObjectCall call)
         {
-            SequenceNumberManager.RecordSequenceNumber(call);
+            int sequenceNumber = SequenceNumberManager.RecordSequenceNumber(call);
+            int last;
+            do
+            {
+                last = this.lastSequenceNumber;
+            }
+            while (sequenceNumber > last &&
+                   sequenceNumber != Interlocked.CompareExchange(ref this.lastSequenceNumber, sequenceNumber, last));

```

Comment:
> I am not embarrassed to tell you that this took me a number of reads until I understood why it worked.

Yeah, it will probably take me a while to understand in a few months. Or maybe even a few days.
I'll add a comment. I should probably extract it to a separate method, too.